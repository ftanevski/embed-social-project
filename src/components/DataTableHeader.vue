<template>
    <div class="row header-row">
        <div class="dropdown">
            <select
                class="form-control"
                v-model="selectedValue"
                @change="updateNumOfPosts"
                data-testid="dropdown-test"
            >
                <option
                    class="select-options"
                    v-for="option in options"
                    :value="option"
                    :key="option"
                    data-testid="dropdown-options-test"
                >
                {{ option }}
                </option>
            </select>
        </div>
        <div class="left-button">
            <button class="btn btn-secondary" @click="openFilterModal">Filter Submissions</button>
        </div>
        <div class="right-button">
            <button class="btn btn-primary">Create Submission</button>
        </div>
    </div>
</template>
 
<script>
export default {
    name: 'DataTableHeader',
    data() {
        return {
            options: [5, 10, 15, 20, 25],
            selectedValue: 5
        }
    },
    emits: [
        'update-posts',
        'filter-modal'
    ],
    methods: {
        updateNumOfPosts() {
            this.$emit('update-posts', this.selectedValue);
        },
        openFilterModal() {
            this.$emit('filter-modal')
        }
    }
}
</script>
 
<style>
.dropdown {
    margin-left: 15px;
}
 
.header-row {
    display: flex;
    margin-bottom: 10px;
}
 
.left-button {
    margin-left: 20px;
}
 
.right-button {
    margin-left: auto;
    margin-right: 15px;
}
</style>