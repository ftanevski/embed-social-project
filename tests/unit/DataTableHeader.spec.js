import { mount } from '@vue/test-utils';
import DataTableHeader from '../../src/components/DataTable';
 
describe('DataTableHeader', () => {
    let wrapper;
 
    beforeEach(() => {
		wrapper = mount(DataTableHeader);
	});
 
    test('Snapshot matches', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
 
    test('Emits the event with the selected value payload', async () => {
        let dropdownSelector = '[data-testid="dropdown-test"]';
        let dropdownOptionsSelector = '[data-testid="dropdown-options-test"]';
 
        wrapper.findAll(dropdownOptionsSelector)[1].element.selected = true;

        await wrapper.find(dropdownSelector).trigger('change');
        let emittedEvents = wrapper.emitted('update-posts');
        expect(emittedEvents).toHaveLength(1);
        console.log(emittedEvents);
    });
})