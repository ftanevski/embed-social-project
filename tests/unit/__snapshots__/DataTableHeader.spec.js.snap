// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeader Snapshot matches 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="row header-row"
  >
    <div
      class="dropdown"
    >
      <select
        class="form-control"
        data-testid="dropdown-test"
      >
        
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="5"
        >
          5
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="10"
        >
          10
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="15"
        >
          15
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="20"
        >
          20
        </option>
        <option
          class="select-options"
          data-testid="dropdown-options-test"
          value="25"
        >
          25
        </option>
        
      </select>
    </div>
    <div
      class="left-button"
    >
      <button
        class="btn btn-secondary"
      >
        Filter Submissions
      </button>
    </div>
    <div
      class="right-button"
    >
      <button
        class="btn btn-primary"
      >
        Create Submission
      </button>
    </div>
  </div>
  <div
    class="col border"
  >
    
    
  </div>
  <div
    class="button-container justify-content-center mt-5"
  >
    <!--v-if-->
  </div>
  
</div>
`;
